src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerizer.java

```diff
@@ -12,23 +12,23 @@
  * Iterator over successive Kmers from a sequence of characters.
  * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
  */
-public class SVKmerizer implements Iterator<SVKmer> {
+public class SVKmerizer<KType extends SVKmerBase> implements Iterator<KType> {
     protected final CharSequence seq;
     protected final int kSize;
     protected int idx = 0;
-    protected SVKmer nextKmer;
+    protected KType nextKmer;
 
-    public SVKmerizer( final byte[] seq, final int kSize ) {
-        this(new ASCIICharSequence(seq), kSize);
+    public SVKmerizer( final byte[] seq, final int kSize, KType kmer ) {
+        this(new ASCIICharSequence(seq), kSize, kmer );
     }
 
-    public SVKmerizer( final CharSequence seq, final int kSize ) {
+    public SVKmerizer( final CharSequence seq, final int kSize, KType kmer ) {
         this.seq = seq;
         this.kSize = kSize;
-        this.nextKmer = nextKmer(new SVKmer(kSize), 0);
+        this.nextKmer = nextKmer(kmer, 0);
     }
 
-    protected SVKmerizer( final int kSize, final CharSequence seq ) {
+    protected SVKmerizer( final int kSize, final CharSequence seq, KType kmer ) {
```

Comment:
This was a result of using generics - Java requires there to be an argument with the generic type. No longer relevant since we're not using generics.